src/block/decompress.rs

```diff
@@ -23,11 +23,29 @@ unsafe fn duplicate(
     {
         duplicate_overlapping(output_ptr, start, match_length);
     } else {
-        crate::block::wild_copy_from_src_16(start, *output_ptr, match_length);
+        debug_assert!(
+            output_ptr.add(match_length / 16 * 16 + ((match_length % 16) != 0) as usize * 16)
+                <= output_end
+        );
+        wild_copy_from_src_16(start, *output_ptr, match_length);
         *output_ptr = output_ptr.add(match_length);
     }
 }
 
+#[inline]
+fn wild_copy_from_src_16(mut source: *const u8, mut dst_ptr: *mut u8, num_items: usize) {
+    // Note: if the compiler transforms this into a call to memcpy it'll hurt performance!
+    // It doesn't seem to be the case though, probably due to the 16 byte stride length.
```

Comment:
really good catch, I also noticed that 16bytes was much faster but thought it was because of the larger register.
I wonder if this is worth reporting as compiler a bug

Reply:
Yeah, misoptimizations like this are tricky. This is likely an intended optimization in LLVM. I'm actually surprised the 16byte loop isn't also being "optimized".

Reply:
I reported it here: https://github.com/rust-lang/rust/issues/84813